widget/view.controller.js

```diff
@@ -96,144 +107,128 @@
       document.getElementsByTagName("head")[0].appendChild(script);
     }
 
+    //fetch live data
     function fetchData() {
       getResourceAggregate().then(function (result) {
         $scope.processing = false;
         if (result && result['hydra:member'] && result['hydra:member'].length > 0) {
-          let idIndex = 1;
-          let nodesMap = {};
-          // Form JSON Data
-          const sourceNodes = _.filter(_.map(result['hydra:member'], item => item.source), (item, index, arr) => item !== null && arr.indexOf(item) == index);
-          const resourceNodes = _.filter(_.map(result['hydra:member'], item => item.target), (item, index, arr) => item !== null && arr.indexOf(item) == index);
-          const relationNodes = _.filter(_.map(result['hydra:member'], item => item.relation), (item, index, arr) => item !== null && arr.indexOf(item) == index);
-          sourceNodes.forEach(node => {
-            nodesMap['s_' + node] = idIndex;
-            nodes.push({
-              "name": node,
-              "id": idIndex++
-            });
-          });
-          resourceNodes.forEach(node => {
-            nodesMap['t_' + node] = idIndex;
-            nodes.push({
-              "name": node,
-              "id": idIndex++
+          let dataToPlot = result['hydra:member'];
+          createDataToPlot(dataToPlot);
+        } else {
+          renderNoRecordMessage('No records found !');
+        }
+      }, function (error) {
+        $scope.processing = false;
+        renderNoRecordMessage('No records found !');
+      }).finally(function () {
+        $scope.processing = false;
+      });
+    }
+
+    function createDataToPlot(dataToPlot) {
```

Comment:
Most of the business logic is added in the view controller. We can move these parts into the service file.